breakpad-symbols/src/lib.rs

```diff
@@ -194,22 +192,25 @@ pub fn relative_symbol_path(module: &dyn Module, extension: &str) -> Option<Stri
 /// Since symbol files can be on the order of a gigabyte(!) and downloaded
 /// from the network, aggressive caching is pretty important. The current
 /// approach is a nice balance of simple and effective.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum SymbolError {
     /// Symbol file could not be found.
     ///
     /// In this case other symbol providers may still be able to find it!
+    #[error("symbol file not found")]
     NotFound,
     /// Symbol file could not be loaded into memory.
-    LoadError(Error),
+    #[error("couldn't read input stream")]
+    LoadError(#[from] std::io::Error),
     /// Symbol file was too corrupt to be parsed at all.
     ///
     /// Because symbol files are pretty modular, many corruptions/ambiguities
     /// can be either repaired or discarded at a fairly granular level
     /// (e.g. a bad STACK WIN line can be discarded without affecting anything
     /// else). But sometimes we can't make any sense of the symbol file, and
     /// you find yourself here.
-    ParseError(Error),
+    #[error("parse error: {0}")]
+    ParseError(String),
```

Comment:
Pretty minor, but given the errors this code is generating in practice, seems like this could be:

```
#[error("parse error: {0} at line {1}")]
ParseError(&'static str, u64)
```

Reply:
Absolutely! I wasn’t aware there is always access to a line number.